src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -143,6 +143,35 @@ public IEnumerable<TagHelperDescriptor> Resolve(TagHelperDescriptorResolutionCon
             return descriptors;
         }
 
+        /// <summary>
+        /// This method returns a deep copy of a given <see cref="TagHelperDescriptor"/>.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/> to clone.</param>
+        /// <returns></returns>
+        public static TagHelperDescriptor CloneDescriptor(TagHelperDescriptor descriptor)
```

Comment:
Would this be better suited as an instance method on `TagHelperDescriptor` i.e. `public TagHelperDescriptor Clone()`? Seems odd to have this be a public method on the resolver.